orchestrator/test_runner/src/erc_721_happy_path.rs

```diff
@@ -0,0 +1,638 @@
+use crate::get_fee;
+use crate::utils::check_erc20_balance;
+use crate::utils::*;
+use crate::MINER_ADDRESS;
+use crate::MINER_PRIVATE_KEY;
+use crate::OPERATION_TIMEOUT;
+use crate::TOTAL_TIMEOUT;
+use bytes::BytesMut;
+use clarity::{Address as EthAddress, Uint256};
+use cosmos_gravity::query::get_attestations;
+use cosmos_gravity::send::send_to_eth;
+use cosmos_gravity::{query::get_oldest_unsigned_transaction_batches, send::send_ethereum_claims};
+use deep_space::address::Address as CosmosAddress;
+use deep_space::coin::Coin;
+use deep_space::private_key::PrivateKey as CosmosPrivateKey;
+use deep_space::Contact;
+use ethereum_gravity::utils::get_valset_nonce;
+use ethereum_gravity::{send_erc721_to_cosmos::send_erc721_to_cosmos, utils::get_tx_batch_nonce};
+use gravity_proto::gravity::query_client::QueryClient as GravityQueryClient;
+use gravity_proto::gravity::MsgSendToCosmosClaim;
+use gravity_proto::gravity::MsgValsetUpdatedClaim;
+use gravity_utils::error::GravityError;
+use gravity_utils::types::SendToCosmosEvent;
+use num::CheckedAdd;
+use prost::Message;
+use std::any::type_name;
+use std::time::Duration;
+use std::time::Instant;
+use tokio::time::sleep as delay_for;
+use tonic::transport::Channel;
+use web30::client::Web3;
+
+pub async fn erc721_happy_path_test(
+    web30: &Web3,
+    grpc_client: GravityQueryClient<Channel>,
+    contact: &Contact,
+    keys: Vec<ValidatorKeys>,
+    gravity_address: EthAddress,
+    erc20_address: EthAddress,
+    validator_out: bool,
+) {
+    info!("This is the ERC 721 Happy path test!");
+    let mut grpc_client = grpc_client;
+
+    let no_relay_market_config = create_default_test_config();
+    start_orchestrators(
+        keys.clone(),
+        gravity_address,
+        validator_out,
+        no_relay_market_config,
+    )
+    .await;
+
+    // bootstrapping tests finish here and we move into operational tests
+
+    // send 3 valset updates to make sure the process works back to back
+    // don't do this in the validator out test because it changes powers
+    // randomly and may actually make it impossible for that test to pass
+    // by random re-allocation of powers. If we had 5 or 10 validators
+    // instead of 3 this wouldn't be a problem. But with 3 not even 1% of
+    // power can be reallocated to the down validator before things stop
+    // working. We'll settle for testing that the initial valset (generated
```

Comment:
this can be removed, we don't need to duplicate that testing.

Reply:
Removed